orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/PublishKeelSpecsTask.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.keel.task
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.KeelService
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.igor.ScmService
+import com.netflix.spinnaker.orca.keel.model.ResourceSpec
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import retrofit.client.Response
+import java.util.concurrent.TimeUnit
+
+@Component
+class PublishKeelSpecsTask
+@Autowired
+constructor(
+  private val keelService: KeelService,
+  private val keelObjectMapper: ObjectMapper,
+  private val scmService: ScmService
+) : RetryableTask {
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  override fun execute(stage: Stage): TaskResult {
+//      parameters:
+//      - name: repoUrl
+//        mapping: cluster.env.MD_REPO_URL
+//        label: Git Repository SSH URL
+//        order: 0
+//      - name: directory
+//        mapping: cluster.env.MD_DIRECTORY
+//        label: File Directory
+//        defaultValue: .netflix/
+//        order: 1
+//      - name: fileExtension
+//        mapping: cluster.env.MD_FILE_EXTENSION
+//        label: File Extension
+//        defaultValue: .yml
+//        order: 2
+//      - name: forceRepublish
+//        mapping: cluster.env.MD_FORCE_REPUBLISH
+//        defaultValue: false
+//        label: Force Republish
+//        order: 3
+//      - name: gitBranch
+//        mapping: cluster.env.GIT_BRANCH
+//        defaultValue: ${trigger.branch}
+//        label: Branch
+//        order: 4
+//      - name: gitSha
+//        mapping: cluster.env.GIT_SHA
+//        defaultValue: ${trigger.hash}
+//        label: SHA
+//        order: 5
+
+    val context = keelObjectMapper.convertValue<PublishKeelSpecsContext>(stage.context)
+    val manifests = scmService.listKeelManifests(
+      context.scmType, context.projectKey, context.repositorySlug, context.ref)
+
+    var failedManifest: String? = null
+    for (manifest in manifests) {
```

Comment:
TODO: this block could be run in parallel for each manifest.

Reply:
We could delete the resources if some fail?

Reply:
We could, but then we need to add extra logic to figure out if the resources existed in keel prior to publishing from here (in which case I don't think we can delete them). Do you feel like that next level of error handling could be part of a separate PR, perhaps?

Reply:
Yes definitely

Reply:
Not applicable anymore since we're refactoring this to publish delivery configs, not resources.